src/main/kotlin/io/curity/identityserver/plugin/signicat/authentication/SignicatCallbackRequestHandler.kt

```diff
@@ -196,30 +196,22 @@ class SignicatCallbackRequestHandler(config : SignicatAuthenticatorPluginConfig)
         val configuration = Properties()
 
         configuration.setProperty("debug", "${!isProd}") // Only enable debug in non-prod
-    
+
         if (isProd)
         {
             configuration.setProperty(ASSERTING_PARTY_DN, PROD_DN)
         }
         else
         {
             logger.trace("SAML response from Signicat: {}", requestModel.samlResponse)
-            
+
             configuration.setProperty(ASSERTING_PARTY_DN, NON_PROD_DN)
         }
-        
+
         val samlFacade = SamlFacade(configuration)
 
         signerTrustCryptoStore.ifPresent { store ->
-            val stream = ByteArrayOutputStream()
-
-            ObjectOutputStream(stream).use { out ->
-                out.writeObject(store.asKeyStore)
-            }
-
-            val bytes = stream.toByteArray()
-            
-            samlFacade.setSamlKeystore(bytes)
+            samlFacade.setSamlKeystore(store.convertToJksByteArray())
```

Comment:
what if the keystore is already a `jks` keystore? Then we do this work for nothing.